packages/playback-core/src/index.ts

```diff
@@ -281,6 +281,43 @@ export const getLiveEdgeStart = (mediaEl: HTMLMediaElement) => {
   return seekable.end(seekable.length - 1) - liveEdgeStartOffset;
 };
 
+const isApproximatelyEqual = (x: number, y: number, moe = 0.001) => Math.abs(x - y) <= moe;
+const isApproximatelyGTE = (x: number, y: number, moe = 0.001) => x > y || isApproximatelyEqual(x, y, moe);
+
+export const isPseudoEnded = (mediaEl: HTMLMediaElement) => {
+  return mediaEl.paused && isApproximatelyGTE(mediaEl.currentTime, mediaEl.duration);
+};
+
+export const getEnded = (mediaEl: HTMLMediaElement, hls?: HlsInterface) => {
+  if (!!hls) return mediaEl.ended;
+  return mediaEl.ended || isPseudoEnded(mediaEl);
+};
+
+export const createPlayPromise = (mediaEl: HTMLMediaElement) => {
+  return new Promise<void>((resolve, reject) => {
+    const seekedListener = () => resolve();
+    const pauseListener = () => {
+      mediaEl.removeEventListener('seeked', seekedListener);
+      reject();
+    };
+    addEventListenerWithTeardown(mediaEl, 'seeked', seekedListener, { once: true });
+    addEventListenerWithTeardown(mediaEl, 'pause', pauseListener, { once: true });
+    mediaEl.currentTime = mediaEl.seekable.start(0);
+  }).then(() => {
+    (muxMediaState.get(mediaEl) ?? {}).pendingPlayPromise = undefined;
+    return mediaEl.play();
+  });
+};
+
+export const getPlayPromise = (mediaEl: HTMLMediaElement, hls?: HlsInterface) => {
+  if (hls || mediaEl.ended || !getEnded(mediaEl, hls)) return mediaEl.play();
+  const mediaState = muxMediaState.get(mediaEl) ?? {};
+  if (!mediaState?.pendingPlayPromise) {
+    mediaState.pendingPlayPromise = createPlayPromise(mediaEl);
+  }
+  return mediaState.pendingPlayPromise;
+};
```

Comment:
I think this play promise logic is not absolutely needed. can it be removed or why is it a hard requirement?